diesel/src/sqlite/connection/stmt.rs

```diff
@@ -58,8 +58,15 @@ impl Statement {
         unsafe { ffi::sqlite3_column_count(self.inner_statement.as_ptr()) as usize }
     }
 
-    /// The lifetime of the returned CStr is shorter than self. This function
-    /// should be tied to a lifetime that ends before the next call to `reset`
+    /// The lifetime of the returned CStr is shorter than self.
+    ///
+    /// > The returned string pointer is valid until either the prepared statement
+    /// > is destroyed by sqlite3_finalize() or until the statement is automatically
+    /// > reprepared by the first call to sqlite3_step() for a particular
+    /// > run or until the next call to sqlite3_column_name()
+    /// > or sqlite3_column_name16() on the same column.
+    /// 
+    /// https://www.sqlite.org/c3ref/column_name.html
     unsafe fn field_name<'a>(&self, idx: usize) -> Option<&'a CStr> {
```

Comment:
The doc states that "The lifetime of the returned `CStr` is shorter than self.", but it looks like this signature actually allows *extending* the lifetime.
```rust
field_name(&self, idx: usize) -> Option<&CStr>
```
or (if we want to be more explicit)
```rust
field_name<'a, 'b: 'a>(&'a self, idx: usize) -> Option<&'b CStr>
```
would solve part of the issue.